### ISSUE TITLE
Various warnings/errors & inability to login

### ISSUE THREAD

**Issue by kraftbj** [rhubarbgroup/redis-cache#226]
Fly-by to see if you have suggestions before I dig into it more tomorrow.

Multinetwork, multisite running 2.0.x latest. Upgraded from 1.6.3. WP trunk.
In WP CLI, getting warnings/errors like this:
```
Notice: Undefined index: [prefix].1:post_meta:61606 in /srv/www/html/wp-content/object-cache.php on line 1362
Warning: array_combine() expects parameter 2 to be array, bool given in /srv/www/html/wp-content/object-cache.php on line 1378
Warning: Invalid argument supplied for foreach() in /srv/www/html/wp-content/object-cache.php on line 1390
```

Additionally, I was logged out and couldn't log back in until disabling or downgrading back to 1.6.3

**Comment by tillkruss** [comment]
Thanks for bringing that up. Looks like the `wp_cache_get_multiple()` implementation isn't working quite right.

**Comment by tillkruss** [comment]
@naxvog: WP 5.5 is slated for 10 August, do you have time to tackle this before then?

**Comment by tillkruss** [comment]
Related cff39fdcc41645305069e82f6949f3476298c0c6.

**Comment by tillkruss** [comment]
Resolved in 6fada41.

**Comment by tillkruss** [comment]
@kraftbj: Would you mind trying https://github.com/rhubarbgroup/redis-cache/releases/tag/2.0.6 ?

**Comment by kraftbj** [comment]
Seeing some PHP warnings still:

`Invalid argument supplied for foreach() - wp-content/object-cache.php:1394`
`array_combine(): Both parameters should have an equal number of elements - wp-content/object-cache.php:1382`

Stack traces:
```
WP_Object_Cache->get_multiple()
wp-content/object-cache.php:128
wp_cache_get_multiple()
wp-includes/functions.php:6382
_get_non_cached_ids()
wp-includes/ms-site.php:353
_prime_site_caches()
wp-includes/class-wp-site-query.php:364
WP_Site_Query->get_sites()
wp-includes/class-wp-site-query.php:254
WP_Site_Query->query()
wp-includes/ms-site.php:464
get_sites()
wp-includes/user.php:685
get_blogs_of_user()
wp-includes/class-wp-admin-bar.php:42
WP_Admin_Bar->initialize()
wp-includes/admin-bar.php:49
_wp_admin_bar_init()
wp-includes/class-wp-hook.php:287
do_action('admin_init')
wp-admin/admin.php:175
```
```
array_combine()
wp-content/object-cache.php:1382
WP_Object_Cache->get_multiple()
wp-content/object-cache.php:128
wp_cache_get_multiple()
wp-includes/functions.php:6382
_get_non_cached_ids()
wp-includes/ms-site.php:353
_prime_site_caches()
wp-includes/class-wp-site-query.php:364
WP_Site_Query->get_sites()
wp-includes/class-wp-site-query.php:254
WP_Site_Query->query()
wp-includes/ms-site.php:464
get_sites()
wp-includes/user.php:685
get_blogs_of_user()
wp-includes/class-wp-admin-bar.php:42
WP_Admin_Bar->initialize()
wp-includes/admin-bar.php:49
_wp_admin_bar_init()
wp-includes/class-wp-hook.php:287
do_action('admin_init')
wp-admin/admin.php:175
```

**Comment by tillkruss** [comment]
Potential fix: 36bd236da7af6c111872bafd54fbaa74efe5defa

**Comment by tillkruss** [comment]
Releases 2.0.7 with a fix to wp.org.
